server/src/controllers/client/getAllClients.js

```diff
@@ -1,9 +1,104 @@
 const logger = require("pino")();
 const Client = require("../../models/client.model");
+const sequelize = require("sequelize");
+const Op = sequelize.Op;
 
 const getAllClientsRequestHandler = async (req, res) => {
   try {
-    const clients = await Client.findAll();
+    const {
+      name,
+      phone_number,
+      date_updated_from,
+      date_updated_to,
+      date_registered_from,
+      date_registered_to,
+      owner,
+      active,
+      r_and_i,
+      closed,
+    } = req.query;
+
+    let query = {};
+    let query_options = {};
+
+    // build the query object
+    if (name) {
+      query = {
+        ...query,
+        [Op.and]: sequelize.literal(`LOWER(name) LIKE :name`),
+      };
+      query_options = {
+        replacements: { name: `%${name.toLowerCase()}%` },
+      };
+    }
+
+    if (phone_number) {
+      query.phone_number = { [Op.like]: `%${phone_number}%` };
+    }
+
+    if (date_updated_from) {
+      query.date_updated = {
+        [Op.gte]: new Date(date_updated_from),
+      };
+    }
+
+    if (date_updated_to) {
+      const date = new Date(date_updated_to);
+      date.setUTCHours(23, 59, 59, 999);
+      query.date_updated = {
+        ...query.date_updated,
+        [Op.lte]: date,
+      };
+    }
+
+    if (date_registered_from) {
+      query.date_added = {
+        [Op.gte]: new Date(date_registered_from),
```

Comment:
same comment as the date_updated_from, to set the hours to the very start of the day (0,0,0,0)